crucible-mir/src/Mir/Trans.hs

```diff
@@ -1112,17 +1118,23 @@ evalPlaceProj ty pl@(MirPlace tpr ref NoMeta) M.Deref = do
         CTyBox t -> doRef t
         _ -> mirFail $ "deref not supported on " ++ show ty
   where
-    doRef (M.TySlice _) | MirSliceRepr tpr' <- tpr = doSlice tpr' ref
-    doRef M.TyStr | MirSliceRepr tpr' <- tpr = doSlice tpr' ref
+    doRef (M.TySlice ty') | MirSliceRepr <- tpr = doSlice ty' ref
+    doRef M.TyStr | MirSliceRepr <- tpr = doSlice (M.TyUint M.B8) ref
     doRef (M.TyDynamic _) | DynRefRepr <- tpr = do
         dynRef <- readMirRef tpr ref
         return $ MirPlaceDynRef dynRef
-    doRef _ | MirReferenceRepr tpr' <- tpr = do
+    doRef ty' | MirReferenceRepr <- tpr = do
+        -- This use of `tyToRepr` is okay because `TyDynamic` and other unsized
+        -- cases are handled above.
+        Some tpr' <- tyToReprM ty'
```

Comment:
```suggestion
        -- This use of `tyToReprM` is okay because `TyDynamic` and other unsized
        -- cases are handled above.
        Some tpr' <- tyToReprM ty'
```